mango-core/src/main/scala/org/bdgenomics/mango/converters/GA4GHutil.scala

```diff
@@ -55,17 +56,23 @@ object GA4GHutil {
    * Converts alignmentRecordRDD to GA4GHReadsResponse string
    *
    * @param alignmentRecordRDD rdd to convert
-   * @return GA4GHReadsResponse json string
+   * @return Map GA4GHReadsResponse json strings for each sample in RDD
    */
-  def alignmentRecordRDDtoJSON(alignmentRecordRDD: AlignmentRecordRDD): String = {
+  def alignmentRecordRDDtoJSON(alignmentRecordRDD: AlignmentRecordRDD): java.util.Map[String, String] = {
     val logger = LoggerFactory.getLogger("GA4GHutil")
 
     val gaReads: Array[ReadAlignment] = alignmentRecordRDD.rdd.collect.map(a => alignmentConverter.convert(a, ConversionStringency.LENIENT, logger))
 
-    val result: ga4gh.ReadServiceOuterClass.SearchReadsResponse = ga4gh.ReadServiceOuterClass.SearchReadsResponse.newBuilder()
-      .addAllAlignments(gaReads.toList.asJava).build()
+    // Group by ReadGroupID, which is set in bdg convert to alignmentRecord's getRecordGroupName(), if it exists, or "1"
+    val results: Map[String, ga4gh.ReadServiceOuterClass.SearchReadsResponse] =
+      gaReads.groupBy(r => r.getReadGroupId).map(sampleReads =>
```

Comment:
@heuermh @jpdna does it make sense to group reads by ReadGroupId? The idea here is to group reads by sample, so when they are written to json they can be fed per sample into some output (ie, to view as separate tracks in a genome browser. ReadGroupId seems to decipher between this in [bdg convert](https://github.com/bigdatagenomics/convert/blob/master/convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/AlignmentRecordToReadAlignment.java#L82)